test/Interop/Cxx/templates/canonical-types.swift

```diff
@@ -0,0 +1,22 @@
+// RUN: %target-run-simple-swift(-I %S/Inputs -Xfrontend -enable-cxx-interop -Xcc -std=c++17)
+//
+// REQUIRES: executable_test
+
+import CanonicalTypes
+import StdlibUnittest
+
+var TemplatesTestSuite = TestSuite("TemplatesTestSuite")
+
+TemplatesTestSuite.test("canonical-types") {
+  // multiple typedeffed types with the same canonical type are the same type
+  // from the typechecking perspective.
```

Comment:
There is a more direct way to write a typechecker test:

```
expectEqualType(WrappedMagicNumberA.self, WrappedMagicNumberB.self)
```

I'm not sure it is valuable to keep the rest of this test (including the SILGen and IRGen counterparts) -- up to you. Do keep the module interface test though.

Reply:
Done.